src/flow.py

```diff
@@ -211,7 +212,11 @@ def _notification_loop(self):
             """Loops calling WaitForNotification on this session."""
             while self.listen_notifications.is_set():
                 try:
-                    changes = self.flow.wait_for_notification(sid=self.sid)
+                    # we need a timeout here because otherwise we will not
+                    # detect that listen_notifications.is_set has been changed
+                    # until we receive another notification, and therefore may
+                    # not shut down promptly when needed.
+                    changes = self.flow.wait_for_notification(sid=self.sid, timeout=15)
```

Comment:
This line has no effect due to: https://github.com/SpiderOak/flow-python/blob/master/src/flow.py#L433.
We force not to timeout on WaitForNotification because of the way WaitForNotification works on the glue. See glue/private/wait_for_notification.go#27
If the client timeouts, then the following may happen:
- client calls `wait_for_notification()`.
- glue waits for notifications in `glue/private/wait_for_notification.go#27`.
- client timeouts the `wait_for_notification()` call.
- Some notification is processed and ack'd in line #27.
- The client is not there to receive those acknowledged notifications.